c/src/sbp.c

```diff
@@ -670,7 +670,7 @@ s8 sbp_all_message_callback_register(sbp_state_t *s,
                                  sbp_msg_callbacks_node_t *node) {
   sbp_callback_t callback;
   callback.decoded = cb;
-  return sbp_register_callback_generic(s, SBP_MSG_ALL, callback,
+  return sbp_register_callback_generic(s, (sbp_msg_type_t) SBP_MSG_ALL, callback,
```

Comment:
The constant `SBP_MSG_ALL` defined [here](https://github.com/swift-nav/libsbp/blob/8ec25d0e9dc8b22403a1313e5c4d8ca81fd63b89/c/include/libsbp/sbp.h#L80-L81) is not part of the enum `sbp_msg_type_t` defined [here](https://github.com/swift-nav/libsbp/blob/8ec25d0e9dc8b22403a1313e5c4d8ca81fd63b89/c/include/libsbp/sbp_msg_type.h#L49-L255). Is this behavior correct?

Reply:
This was intentional, as far as I know, SBP_MSG_ALL is a special/sentinel value for the API, but maybe it would be special cased to be included with the `sbp_msg_type_t` enum-- @woodfell thoughts?

Reply:
And isn't the cast turning it into the first value of the enum `SbpMsgAcqResultDepA`? (assuming the internal representation of the enum matches the first value to 0)

Reply:
SbpMsgAcqResultDepA value is not zero as far as I can tell: https://github.com/swift-nav/libsbp/blob/989d77da0e50a739255e1cfe97418f6e0910d5bc/c/include/libsbp/acquisition_macros.h#L42

Reply:
Oh, you are right!

Reply:
Yeah I think the better option would be to extend the generate so that `SBP_MSG_ALL` (and `SbpMsgAll`) are output rather than casting to an enum value which isn't actually valid